astro-migration-provider/astronomer/migration/env/operators.py

```diff
@@ -0,0 +1,162 @@
+import os
+
+from airflow.models import BaseOperator
+from airflow.utils.session import provide_session
+
+from python_graphql_client import GraphqlClient
+from sqlalchemy.orm import Session
+from typing import Any, Sequence
+
+class AstroEnvMigrationOperator(BaseOperator):
+    """
+    Sends env vars from Airflow to Astronomer Deployment
+    """
+    template_fields: Sequence[str] = ('token', 'deployment_url')
+    ui_color = "#974bde"
+
+    def __init__(
+            self,
+            deployment_url,
+            token,
+            env_include_list,
+            **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.deployment_url = deployment_url
+        self.token = token
+        self.env_include_list = env_include_list
+
+    def _find_deployment_id(self):
+        deployments = self._astro_deployments()
+        for id, deployment in deployments.items():
+            url = deployment['deploymentSpec']['webserver']['url']
+            split_url = url.split("?", 1)
+            base_url = split_url[0]
+            if base_url in self.deployment_url:
+                return id
+                #TODO: Add logic if deployment id is not found
+
+    def _existing_env_vars(self):
```

Comment:
this code should all be shared between the two utilities

Reply:
@fritz-astronomer do we want that to be something that prevents us getting done this week though? We could add it to a backlog and have a working utility now.

Reply:
A ticket is fine by me 👍

Reply:
[Done](https://www.notion.so/astronomerio/Create-shared-modules-between-the-starship-plugin-and-astronomer-migration-provider-bce353a8c1e3482d89a9344f979757c9)